frontend/simple/views/Join.vue

```diff
@@ -0,0 +1,178 @@
+                  </div>
+                  <div class="panel-block">
+                    {{contract.memberApprovalPercentage}}
+                  </div>
+                  <div class="panel-block">
+                    <strong><i18n>Percentage of members are required to remove a member</i18n></strong>
+                  </div>
+                  <div class="panel-block">
+                    {{contract.memberRemovalPercentage}}
+                  </div>
+                  <div class="panel-block">
+                    <strong><i18n>Contribution Transparency</i18n></strong>
+                  </div>
+                  <div class="panel-block">
+                    {{contract.contributionPrivacy}}
+                  </div>
+                </div>
+              </div>
+              <div class="column is-one-quarter">
+                <table class="table is-bordered is-striped is-narrow">
+                <thead>
+                <tr>
+                  <th><i18n>Group Members</i18n></th>
+                </tr>
+                </thead>
+                <tbody>
+                <tr v-for="member in state.members" class="member">
+                  <td>
+                    <div class="media">
+                      <div class="media-left">
+                        <p class="image is-64x64">
+                          <img src="http://bulma.io/images/placeholders/128x128.png">
+                        </p>
+                      </div>
+                      <div class="media-content">
+                        <strong>{{member}}</strong>
+                      </div>
+                    </div>
+                  </td>
+                </tr>
+                </tbody>
+              </table>
+              </div>
+            </div>
+          </div>
+        <div class="column is-1"></div>
+      </div>
+    </section>
+</template>
+<script>
+import * as Events from '../../../shared/events'
+import backend from '../js/backend/'
+import * as db from '../js/database'
+export default {
+  name: 'Join',
+  mounted () {
+    this.fetchData()
+  },
+  methods: {
+    fetchData: async function () {
+      let events = await backend.eventsSince(this.$route.query.groupId, this.$route.query.groupId)
+      let [contract, ...actions] = events.map(e => {
+        return Events[e.entry.type].fromObject(e.entry, e.hash)
+      })
+      let state = contract.toVuexState()
+      actions.forEach(action => {
+        let type = action.constructor.name
+        contract.constructor.vuex.mutations[type](state, action.data)
+        if ((action instanceof Events.RecordInvitation) && (action.data.username === this.$store.state.loggedIn)) {
+          this.inviteHash = action.toHash()
+        }
+      })
+      if (!state.invitees.find(invitee => invitee === this.$store.state.loggedIn)) {
+        console.log(new Error('Invalid Invitation'))
+        this.$router.push({path: '/mailbox'})
+      }
+      this.contract = contract.data
```

Comment:
The shared values, income provided, and maybe all the other parameters can change too. So use the state only, not `contract.data`. Assume everything about the group can change.